src/main/java/racingcar/domain/Cars.java

```diff
@@ -1,39 +1,73 @@
 package racingcar.domain;
 
-import racingcar.dto.CarsResponseDto;
 import racingcar.utils.RandomGenerator;
 
+import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
 import java.util.NoSuchElementException;
+import java.util.function.Function;
 import java.util.stream.Collectors;
 
+import static java.util.stream.Collectors.counting;
+import static java.util.stream.Collectors.groupingBy;
+
 public class Cars {
+    private static final String SAME_NAME_ERROR_MSG_FORMAT = "[ERROR] 중복된 이름이 있습니다 : %s";
+    private static final String COMMA_AND_BLANK = ", ";
+
     private final List<Car> cars;
 
-    public Cars(List<Car> cars) {
-        this.cars = cars;
+    private Cars(List<Car> cars) {
+        validateSameName(cars);
+        this.cars = new ArrayList<>(cars);
     }
 
-    public List<Car> getCars() {
-        return Collections.unmodifiableList(cars);
+    public static Cars createCars(List<Car> cars) {
+        return createCars(cars);
+    }
+
+    public static Cars createCarsByNames(List<String> carNames) {
+        return new Cars(carNames.stream()
+                .map(Car::new)
+                .collect(Collectors.toList()));
+    }
+
+    private void validateSameName(List<Car> cars) {
+        List<String> sameNameList = new ArrayList<>();
+        cars.stream()
+                .map(Car::getName)
+                .collect(groupingBy(Function.identity(), counting()))
+                .forEach((name, count) -> {
+                    if (count > 1) {
+                        sameNameList.add(name.toString());
+                    }
+                });
+        if (sameNameList.size() > 0) {
+            throw new IllegalArgumentException(String.format(SAME_NAME_ERROR_MSG_FORMAT, String.join(COMMA_AND_BLANK, sameNameList)));
+        }
     }
 
     public void moveCars() {
         cars.forEach(car -> car.move(RandomGenerator.generateRandomNumber()));
     }
 
-    public CarsResponseDto findWinners() {
-        int maxPosition = getMaxPosition();
-        return new CarsResponseDto(cars.stream()
-                .filter(car -> car.getPosition() == maxPosition)
+    public GameResult findWinners() {
+        Position maxPosition = getMaxPosition();
+        return new GameResult(cars.stream()
+                .filter(car -> car.isSamePosition(maxPosition))
                 .collect(Collectors.toList()));
     }
 
-    private int getMaxPosition() {
-        return cars.stream()
-                .mapToInt(Car::getPosition)
+    private Position getMaxPosition() {
+        return new Position(cars.stream()
+                .map(Car::getPosition)
+                .mapToInt(Position::index)
                 .max()
```

Comment:
```suggestion
        return cars.stream()
                .map(Car::getPosition)
                .mapToInt(Position::index)
                .max()
                .map(Position::new)
```

생성자 또한 메서드 레퍼런스를 이용할 수 있겠네요 :)

Reply:
넵 맞네요 수정하겠습니다 😀

Reply:
앗 그렇군요!!
mapToInt를 제거하고 Position의 max를 구현하는 방법이 있겠네요!